packages/web-components/fast-foundation/src/disclosure/disclosure.spec.md

```diff
@@ -0,0 +1,98 @@
+# Disclosure
+
+## Overview
+
+A disclosure component is the combination of a button and section of content, a button that toggles the visibility of the section. As defined by the W3C:
```

Comment:
Could this instead potentially use the platform elements of `details` and `summary`? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

Reply:
That's a good idea too. It would require style changes to override the native look and feel. I thought of `details` and `summary`, however, the support for IE is missing.

- https://caniuse.com/details
- https://caniuse.com/summary

Reply:
IE isn't in our matrix and truthfully to leverage the components and styling themselves the existing polyfills won't work. Leaving aside IE, are most of the styling issues something that can be overcome? For near all of our web components, we're fully composing new styles.

I know that one consideration could be the details "arrow". Webkit browsers should support styling of this using the `-webkit-details-marker`. I'm not sure if there's support for Firefox to style/hide that - so that's one thing to consider.

The only other question would be a11y support across browsers. Are you open to doing a quick investigation on the styling bit and whether or not the keyboard interaction matches that proposed by the W3C across major browsers? I'd prefer to use the platform here if there isn't a good reason to recreate it; but if there isn't support in these two areas, I think we can move forward from the idea for now.

Reply:
Okay, I think it should be fine if we don't consider IE.

Yes, I will investigate on Styling and A11Y support across all the browsers.

Reply:
Looks like for styling and A11Y:
- **`summary` should act as a button:** custom implementation
- **`enter` and `space` buttons to toggle:** natively supported in `summary`
- **`list-style-type: none; ` should hide marker in FF:** custom implementation
-  **`details`'s height should be dynamic:** custom implementation

If we go with the `details` and `summary` approach, I think we don't need an invoker slot. `title` attribute should be added for invoker title i.e.` summary` text. @EisenbergEffect, we can also handle animation at CSS level I believe.

We can two variants for `summary` style, 1) as button 2) as link/hypertext

Could you confirm the above specs? After that, I will update both the PRs.

Reply:
We could add a `title` slot that falls back to a `title` attribute. That would make setting a basic title very simple but allow flexibility to slot in custom HTML for the title as well.

```html
<slot name="title">${x => x.title}</slot>
```

Great to hear we could use CSS for the animation.
@chrisdholt may have thoughts on the other items.

Reply:
It sounds like details/summary may be a good path forward here based on the research. Can you update the spec with the above considered and I'll take another pass at it! I think this will move us in a great direction. Thanks!

Reply:
I have updated the specs. Please go through it once and do let me know for further changes if any.

Reply:
I have updated the specs, @chrisdholt. Please go through it once and do let me know for further changes if any.